canals/pipeline/pipeline.py

```diff
@@ -360,7 +360,7 @@ def draw(self, path: Path, engine: RenderingEngines = "mermaid-img") -> None:
             for comp, data in self.graph.nodes(data=True)
         }
         print(sockets)
-        _draw(graph=deepcopy(self.graph), path=path, engine=engine)
+        _draw(graph=networkx.MultiDiGraph(self.graph), path=path, engine=engine)
```

Comment:
Deepcopy replaced with the initialization of a new graph (shallow copy). Less safe in theory, however the draw functions only affect the graph edges, so no harm should be done to the components.